src/odb/include/odb/db.h

```diff
@@ -7520,31 +7594,105 @@ class dbModInst : public dbObject
   static dbSet<dbModInst>::iterator destroy(dbSet<dbModInst>::iterator& itr);
 
   static dbModInst* getModInst(dbBlock* block_, uint dbid_);
-
-  std::string getName() const;
+  dbModITerm* getdbModITerm(dbId<dbModITerm>);
 
   std::string getHierarchicalName() const;
+
+  bool getPinAtIx(unsigned ix, dbModITerm*& ret) const;
+  bool findModITerm(const char* name, dbModITerm*& ret);
+
+  dbSet<dbModITerm> getModITerms();
+
   // User Code End dbModInst
 };
 
+class dbModITerm : public dbObject
+{
+ public:
+  void setFlags(uint flags);
+
+  uint getFlags() const;
+
+  dbModInst* getParent() const;
+
+  void setNet(dbModNet* net);
+
+  dbModNet* getNet() const;
+
+  void setNextNetModiterm(dbModITerm* next_net_moditerm);
+
+  dbModITerm* getNextNetModiterm() const;
+
+  void setPrevNetModiterm(dbModITerm* prev_net_moditerm);
+
+  dbModITerm* getPrevNetModiterm() const;
+
+  void setNextEntry(dbModITerm* next_entry);
+
+  dbModITerm* getNextEntry() const;
+
+  // User Code Begin dbModITerm
+  void setSigType(dbSigType type);
+  dbSigType getSigType();
+  void setIoType(dbIoType type);
+  dbIoType getIoType();
+
+  static dbModITerm* create(dbModInst* parentInstance, const char* name);
+  bool connect(dbModNet*);
+  char* getName();
+  // User Code End dbModITerm
+};
+
+class dbModNet : public dbObject
+{
+ public:
+  dbModule* getParent() const;
+
+  // User Code Begin dbModNet
+  dbSet<dbModITerm> getModITerms();
+  dbSet<dbModBTerm> getModBTerms();
+  dbSet<dbITerm> getITerms();
+  dbSet<dbBTerm> getBTerms();
+
+  static dbModNet* create(dbModule* parentModule, const char* name);
+  const char* getName() const;
+  dbModBTerm* connectedToModBTerm();
+  // User Code End dbModNet
+};
+
 class dbModule : public dbObject
 {
  public:
   const char* getName() const;
 
+  void setModInst(dbModInst* mod_inst);
+
   dbModInst* getModInst() const;
 
   // User Code Begin dbModule
+  std::string getHierarchicalName(std::string& separator);
+  std::string getHierarchicalName() const;
+  std::string hierarchicalNameR(std::string& separator);
+  void highestModWithNetNamed(const char* name,
+                              dbModule*& highest_module,
+                              std::string& highest_net_name);
 
   // Adding an inst to a new module will remove it from its previous
   // module.
   void addInst(dbInst* inst);
 
-  dbSet<dbInst> getInsts();
+  dbBlock* getOwner();
 
   dbSet<dbModInst> getChildren();
+  dbSet<dbModInst> getModInsts();
+  dbSet<dbModNet> getModNets();
+  dbSet<dbModBTerm> getModBTerms();
+  dbSet<dbInst> getInsts();
 
   dbModInst* findModInst(const char* name);
+  void removeModInst(const char* name);
```

Comment:
Why by name?  Shouldn't this take a dbModInst* ?

Reply:
removed